miden-crypto/src/dsa/ecdsa_secp256k1/key_agreement.rs

```diff
@@ -0,0 +1,121 @@
+//! ECDH (Elliptic Curve Diffie-Hellman) key agreement implementation over secp256k1 curve.
+//!
+//! Note that the intended use is in the context of a one-way, sender initiated key agreement
+//! scenario. Namely, when the sender knows the (static) public key of the receiver and it
+//! uses that, together with an ephemeral secret key that it generates, to derive a shared
+//! secret.
+//! This shared secret will then be used to encrypt some message (using for example a key
+//! derivation function).
+//! The public key associated with the ephemeral secret key will be sent alongside the encrypted
+//! message.
```

Comment:
nit: I'd add newlines between lines 6-7 and 8-9. This way, these well render as separate paragraphs in the docs.